static/src/javascripts/projects/common/modules/commercial/creatives/hosted-thrasher.js

```diff
@@ -0,0 +1,41 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/template',
+    'common/modules/commercial/hosted-video',
+    'common/modules/commercial/creatives/add-tracking-pixel',
+    'text!common/views/commercial/creatives/hosted-thrasher.html'
+], function (
+    fastdom,
+    template,
+    hostedVideo,
+    addTrackingPixel,
+    hostedVideoStr
+) {
+    var hostedVideoTpl;
+
+    var HostedThrasher = function ($adSlot, params) {
+        this.$adSlot = $adSlot;
+        this.params = params;
+    };
+
+    HostedThrasher.prototype.create = function () {
+        if (!hostedVideoTpl) {
+            hostedVideoTpl = template(hostedVideoStr);
+        }
+
+        var domPromise = fastdom.write(function () {
+            this.$adSlot.append(hostedVideoTpl({ data: this.params}));
+        }.bind(this));
```

Comment:
It'd be great to figure out a good idiom for a component in general for the codebase - [we did a while ago](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/component.js) <= but that is a terrible execution (I can say so as I did a lot of it).

[Oliver tried another way which is nicer](https://github.com/guardian/frontend/blob/2e00099b6509024fd5a9f04aa7daea03e08281ac/static/src/javascripts/projects/common/utils/create-store.js), but I'm not sure it has buy in from others.

Shall we have a chat about this - or does it potentially fit into docom-future?